File: track/src/main/java/robin/scaffold/track/hook/HookTrack.java
package robin.scaffold.track.hook;

import android.app.Application;
import android.util.Log;

public class HookTrack {
    private static boolean activityLifeCycleRegister = false;

    public static void init(Application application) {
        if (application == null) {
            Log.e("e", "Please init with the param \"Application\"/");
            throw new RuntimeException();
        }
        if (!activityLifeCycleRegister) {
            application.registerActivityLifecycleCallbacks(new HookActivityLifecycleCallbacks());
            activityLifeCycleRegister = true;
        }
    }
}

File: track/src/main/java/robin/scaffold/track/hook/HookTrack.kt
package robin.scaffold.track.hook

import android.app.Application
import android.util.Log

object HookTrack {
    private var activityLifeCycleRegister = false
    fun init(application: Application?) {
        if (application == null) {
            Log.e("e", "Please init with the param \"Application\"/")
            throw RuntimeException()
        }
        if (!activityLifeCycleRegister) {
            application.registerActivityLifecycleCallbacks(HookActivityLifecycleCallbacks())

            activityLifeCycleRegister = true
        }
    }
}